### PR TITLE
Add documentation for credHelper

### PR DESCRIPTION
**Pull request by QiWang19**
Add documentation for credHelper. Close https://github.com/containers/libpod/issues/1675

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

File: docs/containers-auth.json.5.md [hunk 1]
```diff
@@ -37,8 +37,28 @@ their accounts on quay.io and docker.io:
...
+    "auths": {
+        "localhost:5001": {}
+    },
+    "credHelpers": {
+		"registry.example.com": "secretservice"
+	}
+}
+```
+
+For more information on Docker credential stores, please reference the [GitHub docker-credential-helpers project](https://github.com/docker/docker-credential-helpers/releases).
```
Comment (rhatdan) [inline docs/containers-auth.json.5.md:58]: Are these credential helpers not Stores and remove the extra Docker adjective.

File: docs/containers-auth.json.5.md [hunk 2]
```diff
@@ -37,8 +37,28 @@ their accounts on quay.io and docker.io:
...
+
 # SEE ALSO
-    buildah-login(1), buildah-logout(1), podman-login(1), podman-logout(1)
+    buildah-login(1), buildah-logout(1), podman-login(1), podman-logout(1), , skopeo-login(1), skopeo-logout(1)
```
Comment (rhatdan) [inline docs/containers-auth.json.5.md:61]: Extra comma

**Comment by rhatdan** [comment]
@TomSweeneyRedHat PTAL

File: docs/containers-auth.json.5.md [hunk 3]
```diff
@@ -37,8 +37,28 @@ their accounts on quay.io and docker.io:
...
+		"registry.example.com": "secretservice"
+	}
+}
+```
+
+For more information on credential helpers, please reference the [GitHub docker-credential-helpers project](https://github.com/docker/docker-credential-helpers/releases).
+
 # SEE ALSO
-    buildah-login(1), buildah-logout(1), podman-login(1), podman-logout(1)
+    buildah-login(1), buildah-logout(1), podman-login(1), podman-logout(1), skopeo-login(1), skopeo-logout(1)
```
Comment (TomSweeneyRedHat) [inline docs/containers-auth.json.5.md:62]: I know it's old, but we only list podman and bulidah login at the top.  I'd suggest: changing line 15->17 to:

```
The file can have zero to many entries and
is created by a `login` command from a container tool such as `podman login`, 
`buildah login` or `skopeo login`.  
```

File: docs/containers-auth.json.5.md [hunk 4]
```diff
@@ -37,8 +37,28 @@ their accounts on quay.io and docker.io:
 An entry can be removed by using a `logout` command from a container
 tool such as `podman logout` or `buildah logout`.
 
+In addition, credential helpers can be configured for specific registries and the credentials-helper
+software can be used to manage the credentials in a more secure way than depending on the base64 encoded authentication
+provided by `login`.  If the credential helpers are configured for specific registries, the base64 encoded will not be used
+for operations concerning credentials of the specified registries.
+
+When the credential helper is in use on a Linux platform, the auth.json file would contain keys specify the registry domain, and values specify the suffix of the program to use (i.e. everything after docker-credential-).  For example:
```
Comment (TomSweeneyRedHat) [inline docs/containers-auth.json.5.md:45]:
```suggestion
When the credential helper is in use on a Linux platform, the auth.json file would contain keys that specify the registry domain, and values that specify the suffix of the program to use (i.e. everything after docker-credential).  For example:
```
Is there supposed to be a dash at the end of `docker-credential-` in the parens at the end?  I removed it in my suggestion.

Comment (QiWang19) [reply docs/containers-auth.json.5.md]: there should be a dash at the end,  the api will append the config value to this string to look for binary.

File: docs/containers-auth.json.5.md (hunk 3 above)
Comment (QiWang19) [reply docs/containers-auth.json.5.md:62]: fixed

**Comment by rhatdan** [comment]
LGTM

File: docs/containers-auth.json.5.md [hunk 5]
```diff
@@ -37,8 +37,28 @@ their accounts on quay.io and docker.io:
 An entry can be removed by using a `logout` command from a container
 tool such as `podman logout` or `buildah logout`.
 
+In addition, credential helpers can be configured for specific registries and the credentials-helper
+software can be used to manage the credentials in a more secure way than depending on the base64 encoded authentication
+provided by `login`.  If the credential helpers are configured for specific registries, the base64 encoded will not be used
```
Comment (TomSweeneyRedHat) [inline docs/containers-auth.json.5.md:42]: Sorry, missed this last time!
```suggestion
provided by `login`.  If the credential helpers are configured for specific registries, the base64 encoded authentication will not be used
```

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**